weave/trace/autopatch.py

```diff
@@ -4,8 +4,52 @@
 check if libraries are installed and imported and patch in the case that they are.
 """
 
+from typing import Any, Callable, Optional
 
-def autopatch() -> None:
+from pydantic import BaseModel, Field, validate_call
+
+from weave.trace.weave_client import Call
+
+
+class OpSettings(BaseModel):
+    """Op settings for a specific integration.
+    These currently subset the `op` decorator args to provide a consistent interface
+    when working with auto-patched functions.  See the `op` decorator for more details."""
+
+    name: Optional[str] = None
+    call_display_name: Optional[str | Callable[[Call], str]] = None
+    postprocess_inputs: Optional[Callable[[dict[str, Any]], dict[str, Any]]] = None
+    postprocess_output: Optional[Callable[[Any], Any]] = None
+
+
+class IntegrationSettings(BaseModel):
+    """Configuration for a specific integration."""
+
+    enabled: bool = True
+    op_settings: OpSettings = Field(default_factory=OpSettings)
+
+
+class AutopatchSettings(BaseModel):
+    """Settings for auto-patching integrations."""
+
+    anthropic: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    cerebras: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    cohere: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    dspy: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    google_ai_studio: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    groq: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    instructor: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    langchain: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    litellm: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    llamaindex: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    mistral: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    notdiamond: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    openai: IntegrationSettings = Field(default_factory=IntegrationSettings)
+    vertexai: IntegrationSettings = Field(default_factory=IntegrationSettings)
+
+
+@validate_call
```

Comment:
This does type assertions, and helps cast dict to the `AutopatchSettings` model

Reply:
woah! TIL